Docs for 1.0 release (#2915)
kestra-io/docs: content/docs/11.migration-guide/1.0.0/singer-plugin.md

---
title: Singer Tap Plugin Removal
icon: /docs/icons/migration-guide.svg
release: 1.0.0
editions: ["OSS", "EE"]
---

## Overview

> Singer support is being deprecated in Kestra 0.24 and fully removed in Kestra 1.0. This guide walks you through migrating existing Singer pipelines to supported alternatives in Kestra, such as [Airbyte](/plugin/plugin-airbyte), [dlt](/plugin/plugin-dlt), and [CloudQuery](/plugin/plugin-cloudquery).

Author: ```suggestion
Singer support is being deprecated in Kestra 0.24 and fully removed in Kestra 1.0. This guide walks you through migrating existing Singer pipelines to supported alternatives in Kestra, such as [Airbyte](/plugin/plugin-airbyte), [dlt](https://kestra.io/blueprints?page=1&size=24&q=dlt), and [CloudQuery](/plugin/plugin-cloudquery).
```

we don't have a dlt plugin yet, only blueprints

## Why is Singer support being removed?

Singer was once a promising open-source technology for building and sharing data connectors. However:

- It is no longer actively maintained.
- After Meltano shut down, there is no longer a company backing the ecosystem.
- As a result, compatibility, security, and reliability cannot be guaranteed going forward.

> To ensure Kestra users have reliable, well-maintained data ingestion options, we recommend migrating to open-source alternatives such as **Airbyte**, **dlt**, and **CloudQuery**.
> 
> Kestra has first-class plugins for several data ingestion frameworks:
> - Airbyte: Large connector ecosystem for databases, SaaS apps, and warehouses. Runs in both **Cloud** and **OSS** modes.
> - dlt: Flexible **Python-based** ingestion framework, great for custom pipelines and lightweight ingestion.
> - CloudQuery: Purpose-built for cloud asset discovery and syncing metadata to databases or warehouses.

Author: ```suggestion
To ensure Kestra users have reliable, well-maintained data ingestion options, we recommend migrating to other open-source alternatives. For example, Kestra provides plugins for:
- Airbyte: Large connector ecosystem for databases, SaaS apps, and warehouses. Runs in both **Cloud** and **OSS** modes.
- dlt: Flexible **Python-based** ingestion framework, great for custom pipelines and lightweight ingestion.
- CloudQuery: Purpose-built for cloud asset discovery and syncing metadata to databases or warehouses.
```

## Migration Checklist

1. Identify your Singer taps/targets and their data sources/destinations.

2. Check Kestra’s supported plugins for Airbyte, dlt, or CloudQuery equivalents.

3. Configure connections:
    - For Airbyte, set up connections in the Airbyte UI and reference `connectionId` in Kestra.

>     - For dlt, define a Python pipeline and run it as a task in Kestra.

Author: ```suggestion
    - For dlt, define your pipeline script and run it as a Python task in Kestra.
```

>     - For CloudQuery, configure via YAML and execute with the Kestra task.

Author: ```suggestion
    - For CloudQuery, configure your ingestion spec in YAML and execute it with the Sync or CLI tasks.
```
